test/integration/short/metadata-tests.js

```diff
@@ -272,6 +305,40 @@ describe('Metadata', function () {
         }
       ], done);
     });
+    describe('with no callback specified', function () {
+      if (!helper.promiseSupport) {
+        it('should throw an ArgumentError', function (done) {
+          var client = newInstance();
+          utils.series([
+            client.connect.bind(client),
+            function (next) {
+              assert.throws(function () {
+                client.metadata.getTrace(types.Uuid.random());
+              }, errors.ArgumentError);
+              next();
+            },
+            client.shutdown.bind(client)
+          ], done);
+        });
+        return;
+      }
+      it('should return the trace in a promise', function () {
+        var client = newInstance();
+        client.connect()
```

Comment:
Noticed in a few places that the promise isn't being returned from the function, causing mocha to finish the test prematurely, will fix.